### PR TITLE
Privacy/security enhancement

### LINKED ISSUE #228: Garantir dados seguros em campos que usam markdown
### Descrição do Problema
Atualmente os campos que aceita markdown não estão sendo sanitizados.

Implementar códigos que garantam a sanitização de dados para que não estejamos vulneráveis a ataques de injeção.

Campos de interesse:
- Descrição de usuário
- Descrição e título da ação
- Descrição do objetivo
- Descrição de produtos
- Descrição da comunidade
- Comunicados
- Memo da transferência

### PR DESCRIPTION
**Pull request by MatheusBuss**
## What issue does this PR close
Closes no issue

## Changes Proposed ( a list of new changes introduced by this PR)
- [x] Remove potential XSS flaw in rich link controller
- [x] Remove Directory traversal in upload_controller

### CODE REVIEW

**Comment by MatheusBuss** [comment]
@lucca65 this should deal with most of the security concerns raised by sobelow. The only one remaining refers to the upload controller, which I'm not sure how to solve. Maybe we could discuss this one a bit better.

File: lib/cambiatus_web/auth_token.ex [hunk 1]
```diff
@@ -10,7 +10,7 @@ defmodule CambiatusWeb.AuthToken do
 
   @doc "Decodes original data from given `token` and verifies its integrity"
   def verify(token, authorization \\ "user") do
-    Phoenix.Token.verify(Endpoint, auth_salt(authorization), token, max_age: 365 * 24 * 3_600)
+    Phoenix.Token.verify(Endpoint, auth_salt(authorization), token, max_age: 14 * 24 * 3_600)
```
Comment (lucca65) [inline lib/cambiatus_web/auth_token.ex:13]: great!

**Review by lucca65 — CHANGES_REQUESTED** [review]
Bro, this looks great! New tests for auth token, new token invalidation window and an automated tool.

The only thing I think its missing is adding sobelow to our CI, making our builds fail if new security issues arise

**Comment by lucca65** [comment]
Oh i've also removed it from draft state, hope you don't mind. It was already looking ready

**Comment by MatheusBuss** [comment]
> The only thing I think its missing is adding sobelow to our CI, making our builds fail if new security issues arise

Didn't realize that `mix check` was different from the CI :sweat_smile:. Should be working now.

**Review by lucca65 — APPROVED** [review]
Looks like our setup is complete! the build is failing though

**Comment by MatheusBuss** [comment]
It's failling because of the upload controller. I'm fairly certain that we can ignore this one but I wanted to double check it with someone.

It happens because on `lib/cambiatus_web/controllers/upload_controller.ex` - line 14 there's a `File.read!` and this function could be used to read other files in the production machine.

However I don't think that this attack is possible. The file beign read is created by Phoenix as a temp file when an image is uploaded (and the user has no input about this). So this function can only run when a file is uploaded and then Phoenix generates the temp file. The question is if there's no other way this function can be executed. On the tests I ran no code runs if something other than a file is passed to the api.

Another potential concern is [steganography](https://www.sentinelone.com/blog/hiding-code-inside-images-malware-steganography/).

**Comment by lucca65** [comment]
Pois é. O ideal é buscar por outras maneiras de fazer isso. Se não houver, podemos ignorar a analise: https://github.com/nccgroup/sobelow#false-positives

Como isso tá no CI agora, os Merges vão ser bloqueados se não passar. Ou seja, não é possível simplesmente aprovar o PR e ignorar, como dev tens que entregar uma build que é aprovada pelo CI.

**Review by lucca65 — APPROVED** [review]

**Pull request merged**